staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -147,14 +153,44 @@ type Authenticator struct {
 	// The mutex also guards calling the plugin. Since the plugin could be
 	// interactive we want to make sure it's only called once.
 	mu          sync.Mutex
-	cachedToken string
+	cachedCreds *credentials
 	exp         time.Time
+
+	onRotate func()
 }
 
-// WrapTransport instruments an existing http.RoundTripper with credentials returned
-// by the plugin.
-func (a *Authenticator) WrapTransport(rt http.RoundTripper) http.RoundTripper {
-	return &roundTripper{a, rt}
+type credentials struct {
+	token string
+	cert  *tls.Certificate
+}
+
+// UpdateTransportConfig updates the transport.Config to use credentials
+// returned by the plugin.
+func (a *Authenticator) UpdateTransportConfig(c *transport.Config) error {
+	wt := c.WrapTransport
+	c.WrapTransport = func(rt http.RoundTripper) http.RoundTripper {
+		if wt != nil {
+			rt = wt(rt)
+		}
+		return &roundTripper{a, rt}
+	}
+
+	c.TLS.GetCert = a.cert
+
+	defaultDialer := &net.Dialer{Timeout: 30 * time.Second, KeepAlive: 30 * time.Second}
+	// Make a local copy so that we don't cause infinite recursion using
+	// c.Dial->ct.Dial->c.Dial.
+	dial := c.Dial
+	d := connrotation.NewDialer(func(ctx context.Context, network, addr string) (net.Conn, error) {
+		if dial != nil {
```

Comment:
nit: this wouldn't change, right? would it make sense to move this out of this function? e.g.
```go
var dial func(ctx context.Context, network, addr string) (net.Conn, error)
if c.Dial != nil {
  dial = c.Dial
} else {
  dial = &net.Dialer{Timeout: 30 * time.Second, KeepAlive: 30 * time.Second}
}
```

Reply:
Good point. It's a bit uglier than your snipped, but avoids an extra branch after setup